File: MergeTwoSortedLists(21).js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function(list1, list2) {
    let dummy = new ListNode();
    let head = dummy;
    
    while(list1!=null&&list2!=null) {
        head.next = new ListNode();
        head = head.next;
        if(list1.val<list2.val) {
            head.val = list1.val;
            list1 = list1.next;
        } else {
            head.val = list2.val;
            list2 = list2.next;
        }
    }
    
    if(list1==null&&list2!=null) {
        head.next = list2;
    } else if(list1!=null&&list2==null) {
        head.next = list1;
    }
    
    return dummy.next;
};

File: MergeTwoSortedLists(21).swift
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init() { self.val = 0; self.next = nil; }
 *     public init(_ val: Int) { self.val = val; self.next = nil; }
 *     public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }
 * }
 */
class Solution {
    func mergeTwoLists(_ list1: ListNode?, _ list2: ListNode?) -> ListNode? {
        var head1: ListNode? = list1
        var head2: ListNode? = list2
        var sentinel = ListNode()
        var res: ListNode = sentinel
        while head1 != nil && head2 != nil {
            if head1!.val < head2!.val {
                res.next = ListNode(head1!.val)
                head1 = head1!.next
            } else {
                res.next = ListNode(head2!.val)
                head2 = head2!.next
            }
            res = res.next!
        }

        if head1 != nil {
            res.next = head1!
        } else if head2 != nil {
            res.next = head2!
        }

        return sentinel.next
    }
}
